keanu-project/src/main/java/io/improbable/keanu/model/regression/RegressionModelBuilder.java

```diff
@@ -0,0 +1,143 @@
+    }
+
+    /**
+     * Set the input parameters to the distribution describing the prior belief about both the intercept and weights of the regression model
+     *
+     * @param mean The mean of the distribution describing the prior belief about both the regression intercept and weights
+     * @param scaleParameter The scale parameter of the distribution describing the prior belief about both regression intercept and weights.
+     *                       This will represent sigmas if no or ridge regularization is used and will represent betas if lasso regularization is used.
+     * @return this
+     */
+    public RegressionModelBuilder withPriorOnWeightsAndIntercept(double mean, double scaleParameter) {
+        withPriorOnWeights(RegressionWeights.fillPriorOnWeights(this.inputTrainingData.getShape(), mean), RegressionWeights.fillPriorOnWeights(this.inputTrainingData.getShape(), scaleParameter));
+        withPriorOnIntercept(mean, scaleParameter);
+        return this;
+    }
+
+    /**
+     * @return A linear regression model from the data passed to the builder
+     */
+    public RegressionModel<OUTPUT> build() {
+        checkVariablesAreCorrectlyInitialised();
+
+        LinearRegressionGraph<OUTPUT> regressionGraph = new LinearRegressionGraph<>(
+                this.inputTrainingData.getShape(),
+                outputTransform,
+                getInterceptVertex(),
+                getWeightsVertex()
+        );
+
+        performDataFitting(regressionGraph, outputTrainingData);
+        return new RegressionModel<>(regressionGraph);
+    }
+
+    private void checkVariablesAreCorrectlyInitialised() {
+        if (inputTrainingData == null) {
+            throw new IllegalArgumentException("You have not provided input training data");
+        }
+        if (outputTrainingData == null) {
+            throw new IllegalArgumentException("You have not provided output training data");
+        }
+        if (priorOnWeightsMeans == null || priorOnWeightsScaleParameters == null) {
+            withPriorOnWeights(RegressionWeights.fillPriorOnWeights(this.inputTrainingData.getShape(), DEFAULT_MU), RegressionWeights.fillPriorOnWeights(this.inputTrainingData.getShape(), DEFAULT_SCALE_PARAMETER));
+        }
+        if (priorOnInterceptMean == null || priorOnInterceptScaleParameter == null) {
+            withPriorOnIntercept(DEFAULT_MU, DEFAULT_SCALE_PARAMETER);
+        }
+    }
+
+    private DoubleVertex getInterceptVertex() {
+        return this.regularization.getInterceptVertex(priorOnInterceptMean, priorOnInterceptScaleParameter);
+    }
+
+    private DoubleVertex getWeightsVertex() {
+        return this.regularization.getWeightsVertex(getFeatureCount(), priorOnWeightsMeans, priorOnWeightsScaleParameters);
+    }
+
+    private void performDataFitting(LinearRegressionGraph<OUTPUT> regressionGraph, OUTPUT outputTrainingData) {
+        ModelFitter<DoubleTensor, OUTPUT> fitter;
+        switch (this.regularization) {
```

Comment:
You could get rid of this switch statement, and instead have:

    ModelFitter<DoubleTensor, OUTPUT> fitter = this.regularization.createFitterFor(regressionGraph);
    fitter.fit(inputTrainingData, outputTrainingData);